### PR TITLE
Update influx Authorization Headers for write and query path

### PR DESCRIPTION
**Pull request by nhaugo**
- [x] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [x] Rebased/mergable
  - [x] Tests pass
  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

Connect #

### CODE REVIEW

File: server/influx.go [hunk 1]
```diff
@@ -106,10 +107,9 @@ func (h *Service) Write(w http.ResponseWriter, r *http.Request) {
 		req.Host = u.Host
 		req.URL = u
 		// Because we are acting as a proxy, influxdb needs to have the
-		// basic auth information set as a header directly
-		if src.Username != "" && src.Password != "" {
-			req.SetBasicAuth(src.Username, src.Password)
-		}
+		// basic auth or bearer token information set as a header directly
+		auth := influx.DefaultAuthorization(&src)
+		auth.Set(req)
```
Comment (timraymond) [inline server/influx.go]: Unhandled `error` here

File: influx/authorization.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	"fmt"
+	"net/http"
+	"time"
+
+	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/influxdata/chronograf"
+)
+
+// Authorization adds optional authorization header to request
+type Authorization interface {
```
Comment (timraymond) [inline influx/authorization.go]: Mind changing this to an agent noun of some variety to better align with convention (e.g. `io.Writer`, `http.Flusher`, etc.)? Maybe `Authorizer` ?

Comment (goller) [inline influx/authorization.go]: Good point.  Changes incoming!

Comment (goller) [inline influx/authorization.go]: Fixed in 66d2952

**Comment by timraymond** [comment]
Even though it's already merged, I'm still putting in the LGTM gif I had picked out since this is PR #1776

the patriot - imgur

**Pull request merged**